src/Microsoft.DotNet.Tools.Compiler.Native/ArgValues.cs

```diff
@@ -56,8 +57,20 @@ public NativeCompileSettings GetNativeCompileSettings()
 
             if (!string.IsNullOrEmpty(IlcPath))
             {
-                config.IlcPath = IlcPath;
-                config.IlcSdkPath = IlcPath;
+                // We want a directory path. If the user gave us the exact path to the executable
```

Comment:
I'm not sure I agree, but it might because I don't understand precisely what you are proposing. Where do you think the code should belong in NativeCompileSettings?

IMO the purpose of a method such as ArgValues.GetNativeCompileSettings() would be to do transformations (or maybe validation) on the raw arguments provided on the command line before creating the NativeCompileSettings. If the only thing this method were allowed to do is copy fields from ArgValues to NativeCompileSettings without changing them then I would wonder why not get rid of the ArgValues type and just let ArgumentParser.Parse return a new NativeCompileSettings directly?